crates/derive/src/derive_api.rs

```diff
@@ -25,6 +25,8 @@ pub enum Error {
 /// API provides a derivation capability as a cgo::Service.
 pub struct API {
     state: State,
+    // Running tally of the total number of bytes allocated during all invocations of this API.
+    allocated_bytes: i64,
```

Comment:
Yea, this is smelling like instrumentation that should be provided to all through the cgo service.

Reply:
I ripped out all the per-service memory instrumentation, and plan to re-integrate the portions that make sense as part of the instrumentation of stats/logs in the rust side.